torch/autograd/_functions/reduce.py

```diff
@@ -46,7 +46,7 @@ def forward(self, input):
     def backward(self, grad_output):
         if self.dim is None:
             input, = self.saved_tensors
-            grad_input = grad_output.new(self.input_size).fill_(self.result)
+            grad_input = grad_output.new(self.input_size).fill_(self.result * grad_output[0])
```

Comment:
I think this is a GPU sync point also. 
There are many sync points in Functions ( like sum, mean, prod), maybe we can make a different issue for  removing ( after performance testing ) these  points.